### PR TITLE
Add GitHub Actions workflow for main and develop branches

### PR DESCRIPTION
**Pull request by skydoves**
### 🎯 Goal

Add GitHub Actions workflow for main and develop branches.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        with:
+          distribution: adopt
+          java-version: 11
+      - name: Cache Gradle and wrapper
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
```
Comment (JcMinarro) [inline .github/workflows/build.yml:24]: Can't we use our own `.github/actions/gradle-cache` action?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+name: Android CI
+
+on:
+  push:
+    branches: [ main, develop ]
+
+jobs:
+  build:
```
Comment (JcMinarro) [inline .github/workflows/build.yml:8]: This job doesn't do anything.
It only changes make `./gradlew` executable (It should already), but it is not used at all.
The rest of the jobs will start their own container and this change is not used

Comment (JcMinarro) [inline .github/workflows/build.yml:8]: This job doesn't do anything.
It only changes make `./gradlew` executable (It should already), but it is not used at all.
The rest of the jobs will start their own container and this change is not used

Comment (JcMinarro) [inline .github/workflows/build.yml:8]: This job doesn't do anything.
It only changes make `./gradlew` executable (It should already), but it is not used at all.
The rest of the jobs will start their own container and this change is not used

File: .github/workflows/build.yml (hunk 1 above)
Comment (skydoves) [reply .github/workflows/build.yml]: Updated: [570cb52](https://github.com/GetStream/stream-log/pull/11/commits/570cb5284601c8c10105b000d33a08b7fc762c32)

File: .github/workflows/build.yml (hunk 2 above)
Comment (skydoves) [reply .github/workflows/build.yml]: Updated: [570cb52](https://github.com/GetStream/stream-log/pull/11/commits/570cb5284601c8c10105b000d33a08b7fc762c32)

**Comment by skydoves** [comment]
I just updated the workflow following our [build-and-test](https://github.com/GetStream/stream-chat-android/blob/main/.github/workflows/build-and-test.yml) Stream chat workflow.

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**